cloudbridge/cloud/providers/gce/provider.py

```diff
@@ -73,18 +74,10 @@ def gce_compute(self):
 
     def _connect_gce_compute(self):
         if self.credentials_file:
-            with open(self.credentials_file) as f:
-                data = json.load(f)
-                credentials = SignedJwtAssertionCredentials(
-                    data['client_email'], data['private_key'],
-                    'https://www.googleapis.com/auth/compute')
+            credentials = GoogleCredentials.from_json(self.credentials_file)
```

Comment:
I couldn't get this branch of the code to work. I trid providing the credentials file as a bytearray, json object, and a filepath - but it complained on each go. I managed to get the get_application_default() branch to work by exporting the GOOGLE_APPLICATION_CREDENTIALS env var.